app/src/main/kotlin/com/github/gotify/settings/SettingsActivity.kt

```diff
@@ -107,6 +120,36 @@ internal class SettingsActivity : AppCompatActivity(), OnSharedPreferenceChangeL
             }
         }
 
+        override fun onResume() {
+            super.onResume()
+            checkSystemAlertWindowPermission()
+        }
+
+        private fun openSystemAlertWindowPermissionPage(): Boolean {
+            Intent(
+                Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+                Uri.parse("package:${requireContext().packageName}")
+            ).apply {
+                startActivity(this)
+            }
+            return true
+        }
+
+        private fun checkSystemAlertWindowPermission() {
+            findPreference<SwitchPreferenceCompat>(
+                getString(R.string.setting_key_intent_dialog_permission)
+            )?.let {
+                val canDrawOverlays = Settings.canDrawOverlays(requireContext())
+                it.isEnabled = !canDrawOverlays
```

Comment:
Let's remove this line so the user can easily revoke the permissions.

Reply:
As you wish 🫡